internal/worker/sshserver/worker.go

```diff
@@ -115,9 +116,22 @@ func (ssw *serverWrapperWorker) loop() error {
 
 	port := config.SSHServerPort()
 
+	// TODO(ale8k): As of right now, the generated host key is in mongo.
+	// The initialisation logic needs migrating over to DQLite and then
+	// a domain service should call the method to retrieve the generated
+	// host key here. For now, we'll generate it to stop the server bouncing.
+	//
+	// We could access state directly in the worker here, but 4.0 isn't
+	// released yet anyway.
+	hostKey, err := pkissh.NewMarshalledED25519()
+	if err != nil {
+		return errors.Annotatef(err, "failed to ensure ssh server host key")
```

Comment:
This will cause the worker to bounce constantly. 

You can return `dependency.ErrUninstall`, which will cause the worker to uninstall itself. It won't come back until the next reboot.

Reply:
I wonder if this could be done in the manifold, so we don't even spawn anything if we know it's not going to work.

Reply:
Returning `ErrUninstall` will mess with the worker's tests.

You could change the manifold start func to return a `ValueWorker` instead of the SSH worker, with a comment to swap it back later.

Reply:
This will never fail and is genuinely an interim until we wire it up properly. Is this not OK under those conditions for now?

Reply:
> Returning ErrUninstall will mess with the worker's tests.

So? I'd rather mess with the tests than have it bounce while trying to run it.

Reply:
@manadart @ale8k I think the best approach is to remove it from the agent manifold.

Reply:
It can make it a pain when forward-merging because we'll need to plug and unplug for Q/A each time.

Reply:
See [this comment](https://github.com/juju/juju/pull/19190#issuecomment-2740028615) please.